### PR TITLE
Small refactoring to aid approachability

### PR DESCRIPTION
**Pull request by dazoakley**
- Separate out nested classes in non-obvious locations.
- Move spec files around so it's a little more clear what they're representing.
- Introduce a couple of module files to load child classes.
- Make it so consuming code only needs to require "alephant/broker" to get going.

### CODE REVIEW

File: lib/alephant/broker.rb [hunk 1]
```diff
@@ -10,42 +10,11 @@ def self.handle(load_strategy, env)
...
-      def send(response)
-        [
-          response.status,
-          response.headers,
-          [
-            response.content.to_s
-          ]
-        ]
-      end
+      @configuration = c
```
Comment (Integralist) [inline lib/alephant/broker.rb]: Will this work with a class level value?

**Comment by Integralist** [comment]
One comment, but otherwise happy with the refactor

File: lib/alephant/broker.rb (hunk 1 above)
Comment (dazoakley) [inline lib/alephant/broker.rb]: yup, as it's a module there is only one "instance" of it, so no need to the `@@` variable.

**Comment by dazoakley** [comment]
@Integralist you happy for this to go in?

**Comment by Integralist** [comment]
👍

**Pull request closed without merging**